clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/UpsertGoogleScalingPolicyAtomicOperation.groovy

```diff
@@ -0,0 +1,124 @@
+
+class UpsertGoogleScalingPolicyAtomicOperation implements AtomicOperation<Void> {
+  private static final String BASE_PHASE = "UPSERT_SCALING_POLICY"
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  @Autowired
+  private GoogleOperationPoller googleOperationPoller
+
+  @Autowired
+  private GoogleClusterProvider googleClusterProvider
+
+  private final UpsertGoogleScalingPolicyDescription description
+
+  UpsertGoogleScalingPolicyAtomicOperation(UpsertGoogleScalingPolicyDescription description) {
+    this.description = description
+  }
+
+  /**
+   * curl -X POST -H "Content-Type: application/json" -d '[ { "upsertScalingPolicy": { "serverGroupName": "my-zonal-server-group", "region": "us-central1", "credentials": "my-google-account", "autoscalingPolicy": { "coolDownPeriodSec": 60, "cpuUtilization": { "utilizationTarget": 0.9 }, "maxNumReplicas": 3, "minNumReplicas": 2} } } ]' localhost:7002/gce/ops
+   * curl -X POST -H "Content-Type: application/json" -d '[ { "upsertScalingPolicy": { "serverGroupName": "my-regional-server-group", "region": "us-central1", "credentials": "my-google-account", "autoscalingPolicy": { "coolDownPeriodSec": 60, "cpuUtilization": { "utilizationTarget": 0.9 }, "loadBalancingUtilization": { "utilizationTarget" : 0.6 }, "maxNumReplicas": 3, "minNumReplicas": 2, "customMetricUtilizations" : [ { "metric": "myMetric", "utilizationTarget": 0.9, "utilizationTargetType" : "DELTA_PER_SECOND" } ] } } } ]' localhost:7002/gce/ops
+  */
+
+  @Override
+  Void operate(List priorOutputs) {
+    def serverGroupName = description.serverGroupName
+    task.updateStatus BASE_PHASE, "Initializing upsert of scaling policy for $serverGroupName..."
+
+    def credentials = description.credentials
+    def project = credentials.project
+    def compute = credentials.compute
+    def accountName = description.accountName
+    def region = description.region
+    def serverGroup = GCEUtil.queryServerGroup(googleClusterProvider, accountName, region, serverGroupName)
+    def isRegional = serverGroup.regional
+    def zone = serverGroup.zone
+    def location = isRegional ? region : zone
+    def autoscaler = buildAutoscaler(credentials, project, location, serverGroupName, isRegional)
+
+    task.updateStatus BASE_PHASE, "Attempting to retrieve existing autoscaler for $serverGroupName..."
+
+    def scalingPolicies
+    if (isRegional) {
+      scalingPolicies = compute.regionAutoscalers().list(project, region).execute().getItems()
+    } else {
+      scalingPolicies = compute.autoscalers().list(project, zone).execute().getItems()
+    }
+
+    def scalingPolicyExistsForMIG = scalingPolicies.any { it.name == serverGroupName }
+
+    def operation
+    if (scalingPolicyExistsForMIG) {
+      task.updateStatus BASE_PHASE, "Updating existing autoscaler $serverGroupName..."
+
+      if (isRegional) {
+        operation = compute.regionAutoscalers().update(project, region, autoscaler).execute()
```

Comment:
Can't we still make individual updates by just re-composing the entire autoscaler?
We do something similar in `CopyLastGoogleServerGroupAtomicOperation` by first retrieving the full existing representation, and then overriding just the explicitly-passed properties. This would enable what I think is an important use case of just specifying new min/max values, but nothing else. This will make our lives easier when we get the full Resize/Autoscaler relationship sorted.